make/Program.mk

```diff
@@ -73,12 +73,25 @@ flash: all test_softdevice flash_mbr
 ifdef ID
 	$(Q)printf "w4 $(ID_FLASH_LOCATION), 0x$(ID_SECON) 0x$(ID_FIRST)\n" >> $(BUILDDIR)flash.jlink
 endif
-ifeq ($(USE_BOOTLOADER),1)
+ifeq ($(strip $(grep -q Microsoft /proc/version)),)
+	@echo Flashing: $(HEX)
+	nrfjprog.exe -f nrf52 --program $(HEX) --sectorerase
+	nrfjprog.exe -f nrf52 --reset
+else ifeq ($(USE_BOOTLOADER),1)
 	$(Q)printf "loadfile $(MERGED_HEX) \nr\ng\nexit\n" >> $(BUILDDIR)flash.jlink
-else
+	$(Q)$(JLINK) $(JLINK_FLAGS) $(BUILDDIR)flash.jlink
+else ifeq ($(USE_BOOTLOADER),0)
 	$(Q)printf "loadfile $(HEX) \nr\ng\nexit\n" >> $(BUILDDIR)flash.jlink
-endif
 	$(Q)$(JLINK) $(JLINK_FLAGS) $(BUILDDIR)flash.jlink
+endif
+	
+
+flashWSL: all test_softdevice flash_mbr
+	@echo Flashing: $(OUTPUT_DIRECTORY)/toxophilite.hex
```

Comment:
This seems specific to a single application

Reply:
Good catch. Resolved.